packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -76,39 +78,66 @@ def initialize_settings(self):
+            "chat_history": self.settings["chat_history"],
+            "root_dir": self.serverapp.root_dir,
+            "dask_client_future": dask_client_future,
+            # TODO: Set ask_chat_handler based on a retriever related to the learn_chat_handler
+            "retriever": None,
         }
-        default_chat_handler = DefaultChatHandler(
-            **chat_handler_kwargs, chat_history=self.settings["chat_history"]
-        )
-        clear_chat_handler = ClearChatHandler(
-            **chat_handler_kwargs, chat_history=self.settings["chat_history"]
-        )
-        generate_chat_handler = GenerateChatHandler(
-            **chat_handler_kwargs,
-            root_dir=self.serverapp.root_dir,
-        )
-        learn_chat_handler = LearnChatHandler(
-            **chat_handler_kwargs,
-            root_dir=self.serverapp.root_dir,
-            dask_client_future=dask_client_future,
-        )
-        help_chat_handler = HelpChatHandler(**chat_handler_kwargs)
+
+        for chat_handler_ep in chat_handler_eps:
+            try:
+                chat_handler = chat_handler_ep.load()
+            except Exception as err:
+                self.log.error(
+                    f"Unable to load chat handler class from entry point `{chat_handler_ep.name}`: "
+                    + f"Unexpected {err=}, {type(err)=}"
+                )
+                continue
+
+            # Each slash ID, including None (default), must be used only once.
+            # Slash IDs may contain only alphanumerics and underscores.
+            slash_id = chat_handler.slash_id
+
+            if slash_id:
+                # TODO: Validate slash ID (/^[A-Za-z0-9_]+$/)
+                command_name = f"/{slash_id}"
+            else:
+                command_name = "default"
+            self.log.info(
+                f"Trying to register chat handler `{chat_handler.id}` with command `{command_name}`"
+            )
+
+            if command_name in jai_chat_handlers:
+                self.log.error(
+                    f"Unable to register chat handler `{chat_handler.id}` because command `{command_name}` already has a handler"
+                )
+                continue
+
+            jai_chat_handlers[command_name] = chat_handler
+            self.log.info(
+                f"Registered chat handler `{chat_handler.id}` with command `{command_name}`."
+            )
+
+        self.settings["jai_chat_handlers"] = jai_chat_handlers
+
+        """
         retriever = Retriever(learn_chat_handler=learn_chat_handler)
         ask_chat_handler = AskChatHandler(**chat_handler_kwargs, retriever=retriever)
-        self.settings["jai_chat_handlers"] = {
-            "default": default_chat_handler,
-            "/ask": ask_chat_handler,
-            "/clear": clear_chat_handler,
-            "/generate": generate_chat_handler,
-            "/learn": learn_chat_handler,
-            "/help": help_chat_handler,
-        }
-
+        """
```

Comment:
Yeah we definitely need to investigate how we handle this interdependency, but we can do that in the next iteration.